isic-archive-tasks/isic_archive_tasks/image.py

```diff
@@ -0,0 +1,199 @@
+            'large-image': None,
+            'superpixel-mask': None
+        }
+        Image().save(image)
+
+        callback = markImageIngested.s(image['_id'])
+        chord([generateSuperpixels.s(image['_id']),
+               generateLargeImage.s(image['_id'])])(callback)
+    except Exception:
+        sentry_sdk.capture_exception()
+        image['readable'] = False
+        image['ingested'] = True
+        Image().save(image)
+
+
+@app.task(base=CredentialedGirderTask)
+def markImageIngested(results, imageId):
+    from girder.plugins.isic_archive.models.image import Image
+    image = Image().load(imageId, force=True)
+    image['ingested'] = True
+    Image().save(image)
+
+
+@app.task(bind=True, base=CredentialedGirderTask)
+def generateSuperpixels(self, imageId):
+    try:
+        from girder.plugins.isic_archive.models.image import Image
+        from girder.plugins.isic_archive.models.segmentation_helpers.scikit import \
+            ScikitSegmentationHelper
+        image = Image().load(imageId, force=True)
+        imageFile = Image().originalFile(image)
+
+        originalFileStreamResponse = requests.get(
+            'http://isic-archive.test/api/v1/file/%s/download' % imageFile['_id'], headers={
+                'Girder-Token': self.token
+            })
+        originalFileStreamResponse.raise_for_status()
+        originalFileStreamResponse = io.BytesIO(originalFileStreamResponse.content)
+
+        # Scikit-Image is ~70ms faster at decoding image data
+        originalImageData = ScikitSegmentationHelper.loadImage(originalFileStreamResponse)
+
+        superpixelsData = ScikitSegmentationHelper.superpixels(originalImageData)
+        superpixelsEncodedStream = ScikitSegmentationHelper.writeImage(
+            superpixelsData, 'png')
+
+        uploadSuperpixelsResponse = requests.post('http://isic-archive.test/api/v1/file', params={
+            'parentType': 'item',
+            'parentId': imageId,
+            'name': '%s_superpixels_v%s.png' % (image['name'], SUPERPIXEL_VERSION),
+            'size': len(superpixelsEncodedStream.getvalue())
+        }, data=superpixelsEncodedStream.getvalue(), headers={
+            'Girder-Token': self.token
+        })
+        uploadSuperpixelsResponse.raise_for_status()
+
+        superpixelsFile = File().load(uploadSuperpixelsResponse.json()['_id'], force=True)
+        image = Image().load(imageId, force=True, exc=False)
+
+        superpixelsFileNameMatch = re.match(
+            r'^%s_superpixels_v([0-9.]+)\.png' % image['name'],
+            superpixelsFile['name'])
+        if not superpixelsFileNameMatch:
+            return
+
+        superpixelsVersion = float(superpixelsFileNameMatch.group(1))
+        superpixelsFile['superpixelVersion'] = superpixelsVersion
+        # Work around an upstream Girder bug where "File().validate" sets
+        #   superpixelsFile['exts'] = ['0', 'png']
+        superpixelsFile = File().validate(superpixelsFile)
```

Comment:
I'm not sure why we need to validate again, we can discuss more if you can't figure it out, but probably remove.